vms fail to import when start order configured

**Describe the bug**
vms fail to import when start order configured. Containers work correctly.
I saw there was a feature enhancement to add order support, but I'm not asking for that here, just that the order be ignored if not going to process it. This is an api handling failure that should fail gracefully if options not supported.

**To Reproduce**
Steps to reproduce the behavior:
Attempt to import a VM with a start order option configured

Please also provide a minimal Terraform configuration that reproduces the issue.

**Expected behavior**
Unsupported options be ignored instead of causing parsing errors

**Screenshots**
╷
│ Error: error retrieving VM: failed to decode HTTP GET response (path: nodes/apathy/qemu/105/config) - Rea
son: json: cannot unmarshal string into Go struct field GetResponseData.data.startup of type vms.CustomStar
tupOrder
│


**Additional context**
Add any other context about the problem here.
- Provider version (ideally it should be the latest version): 0.24.0
- Terraform version: v1.5.1
- OS (where you run Terraform from)): multiple
- Debug logs (`TF_LOG=DEBUG terraform apply`):

2023-07-10T19:56:27.761-0500 [DEBUG] provider.terraform-provider-proxmox_v0.24.0.exe: Received HTTP Respons
e: Date="Tue, 11 Jul 2023 00:56:28 GMT" tf_http_res_status_code=200 tf_resource_type=proxmox_virtual_enviro
nment_vm tf_rpc=ReadResource @module=proxmox Content-Length=521 Pragma=no-cache Server=pve-api-daemon/3.0 t
f_http_op_type=response tf_http_res_body={"data":{"name":"kadmin1","ostype":"l26","vmgenid":"89ba60cd-7685-
4c7d-abc3-59f707c90852","startup":"order=500","cores":2,"cpu":"host","boot":"order=scsi0;ide2;net0","agent"
:"1","meta":"creation-qemu=6.2.0,ctime=1660774648","onboot":1,"sockets":1,"scsi0":"tank:vm-105-disk-0,size=
32G","memory":2548,"ide2":"none,media=cdrom","numa":1,"net0":"virtio=9E:4D:F1:EB:12:A6,bridge=vmbr0,tag=2",
"digest":"eb266642df7c6fc4df9b7271908f33d86fe4403c","smbios1":"uuid=ff87fae9-2f84-44ed-91cf-48f7520c49ba","
scsihw":"virtio-scsi-pci"}} tf_http_res_status_reason="200 OK" tf_mux_provider=tf5to6server.v5tov6Server Co
nnection=Keep-Alive tf_req_id=75d3fd51-5ad8-183a-318c-3b143cf16a9b Content-Type=application/json;charset=UT
F-8 Expires="Tue, 11 Jul 2023 00:56:28 GMT" tf_http_trans_id=c012c4ff-1651-8e9e-3427-138c4a76d4c3 tf_provid
er_addr=registry.terraform.io/bpg/proxmox Cache-Control=max-age=0 tf_http_res_version=HTTP/1.1 @caller=gith
ub.com/hashicorp/terraform-plugin-sdk/v2@v2.27.0/helper/logging/logging_http_transport.go:162 timestamp=202
3-07-10T19:56:27.761-0500
2023-07-10T19:56:27.764-0500 [ERROR] provider.terraform-provider-proxmox_v0.24.0.exe: Response contains err
or diagnostic: tf_provider_addr=registry.terraform.io/bpg/proxmox tf_resource_type=proxmox_virtual_environm
ent_vm @caller=github.com/hashicorp/terraform-plugin-go@v0.18.0/tfprotov6/internal/diag/diagnostics.go:58 d
iagnostic_severity=ERROR tf_proto_version=6.3 tf_req_id=75d3fd51-5ad8-183a-318c-3b143cf16a9b tf_rpc=ReadRes
ource @module=sdk.proto diagnostic_detail= diagnostic_summary="error retrieving VM: failed to decode HTTP G
ET response (path: nodes/apathy/qemu/105/config) - Reason: json: cannot unmarshal string into Go struct fie
ld GetResponseData.data.startup of type vms.CustomStartupOrder" timestamp=2023-07-10T19:56:27.764-0500
2023-07-10T19:56:27.764-0500 [ERROR] vertex "import proxmox_virtual_environment_vm.kadmin1 result" error: e
rror retrieving VM: failed to decode HTTP GET response (path: nodes/apathy/qemu/105/config) - Reason: json:
 cannot unmarshal string into Go struct field GetResponseData.data.startup of type vms.CustomStartupOrder
2023-07-10T19:56:27.765-0500 [ERROR] vertex "proxmox_virtual_environment_vm.kadmin1 (import id \"apathy/105
\")" error: error retrieving VM: failed to decode HTTP GET response (path: nodes/apathy/qemu/105/config) -
Reason: json: cannot unmarshal string into Go struct field GetResponseData.data.startup of type vms.CustomS
tartupOrder
2023-07-10T19:56:27.765-0500 [ERROR] vertex "proxmox_virtual_environment_vm.kadmin1 (expand)" error: error
retrieving VM: failed to decode HTTP GET response (path: nodes/apathy/qemu/105/config) - Reason: json: cann
ot unmarshal string into Go struct field GetResponseData.data.startup of type vms.CustomStartupOrder

--- a/proxmox/nodes/vms/vms_types.go
+++ b/proxmox/nodes/vms/vms_types.go
@@ -1696,6 +1696,46 @@ func (r *CustomSMBIOS) UnmarshalJSON(b []byte) error {
 	return nil
 }
 
+// UnmarshalJSON converts a CustomStartupOrder string to an object.
+func (r *CustomStartupOrder) UnmarshalJSON(b []byte) error {
+	var s string
+
+	if err := json.Unmarshal(b, &s); err != nil {
+		return fmt.Errorf("failed to unmarshal CustomStartupOrder: %w", err)
+	}
+
+	pairs := strings.Split(s, ",")
+
+	for _, p := range pairs {
+		v := strings.Split(strings.TrimSpace(p), "=")
+
+		if len(v) == 2 {
+			switch v[0] {
+			case "order":
+				order, err := strconv.Atoi(v[1])
+				if err != nil {
+					return fmt.Errorf("failed to parse int: %w", err)
+				}
+				r.Order = &order
+			case "up":
+				up, err := strconv.Atoi(v[1])
+				if err != nil {
+					return fmt.Errorf("failed to parse int: %w", err)
+				}
+				r.Up = &up
+			case "down":
+				down, err := strconv.Atoi(v[1])
+				if err != nil {
+					return fmt.Errorf("failed to parse int: %w", err)
+				}
+				r.Down = &down
+			}
+		}
+	}
+
+	return nil
+}
+
 // UnmarshalJSON converts a CustomStorageDevice string to an object.
 func (r *CustomStorageDevice) UnmarshalJSON(b []byte) error {
 	var s string

--- a/proxmox/nodes/vms/vms_types.go
+++ b/proxmox/nodes/vms/vms_types.go
@@ -1716,18 +1716,21 @@ func (r *CustomStartupOrder) UnmarshalJSON(b []byte) error {
 				if err != nil {
 					return fmt.Errorf("failed to parse int: %w", err)
 				}
+
 				r.Order = &order
 			case "up":
 				up, err := strconv.Atoi(v[1])
 				if err != nil {
 					return fmt.Errorf("failed to parse int: %w", err)
 				}
+
 				r.Up = &up
 			case "down":
 				down, err := strconv.Atoi(v[1])
 				if err != nil {
 					return fmt.Errorf("failed to parse int: %w", err)
 				}
+
 				r.Down = &down
 			}
 		}

